upgrades/upgrade_testing_framework/steps/step_perform_post_upgrade_test.py

```diff
@@ -1,22 +1,25 @@
 from upgrade_testing_framework.core.framework_step import FrameworkStep
 from robot_tests.test_executor import TestExecutor
 
-class PostUpgradeRobotTest(FrameworkStep):
+class PerformPostUpgradeTest(FrameworkStep):
     """
     This step is where you run post-upgrade tests
     """
 
     def _run(self):
         # Get the state we need
         target_cluster = self.state.target_cluster
+        port = target_cluster.rest_ports[0]
+        expectations = self.state.eligible_expectations
+        output_directory = f"{self.state.get_key('test_results_directory')}/post_upgrade"
```

Comment:
Also not a blocker for this PR, but I would propose that this section of a `_run()` body be used *solely* to pull global state into local variables, without any changes or modifications.  The premise is to make an incredibly clear boundary between the FrameworkStep and he global state it interacts with.

Going forward, I'd like to impose this as a non-negotiable convention.  @mikaylathompson, FYI.

Reply:
A little inconvenient but probably a good convention to have as this does create some ambiguity with what the state file has versus what was actually used. Will update with next PR

Reply:
I agree with that as a convention. I wonder if there's a good way to enforce/check for it down the road, but for now, agreed that we should self-impose it.